source/agora/test/NetworkManager.d

```diff
@@ -75,22 +76,44 @@ unittest
             auto prev_key = () @trusted { return KeyPair.random(); }();
 
             Block last_block;
-            // make 20 blocks which have an invalid previous hash
-            foreach (idx; 0 .. 20)
+            // make 3 blocks which have an invalid previous hash
+
+            auto txs = genesisSpendable().map!(txb => txb.sign()).array();
+            auto block = makeNewBlock(last_block, txs);
+            blocks ~= block;
+            last_tx = txs;
+            last_block = block;
+
+            foreach (idx; 1 .. 3)
             {
-                auto txs = makeChainedTransactions(prev_key, last_tx, 1);
+                txs = last_tx.map!(tx => TxBuilder(tx).sign()).array();
                 last_tx = txs;
-                auto block = makeNewBlock(last_block, txs);
+                block = makeNewBlock(last_block, txs);
 
                 blocks ~= block;
                 last_block = block;
             }
 
+            auto signTx (Transaction tx) @trusted { return prev_key.secret.sign(hashFull(tx)[]); }
+
+            foreach (idx1, txs1; blocks)
+            {
+                foreach (tx; blocks[idx1].txs)
+                {
+                    iota(tx.outputs.length).map!(
+                        idx => tx.outputs[idx].address = prev_key.address);
+                    iota(tx.inputs.length).map!(
+                        idx => tx.inputs[idx].signature = signTx(blocks[idx1].txs[idx]));
+                }
+            }
+
+            iota(blocks.length).map!(idx => blocks[idx].header.merkle_root = blocks[idx].buildMerkleTree());
```

Comment:
Some more things, Plz.
```suggestion
            blocks.map!(block => block.header.merkle_root = block.buildMerkleTree());
```

Reply:
@linked0 : `map` should not be used to modify the input. As @AndrejMitrovic pointed out, this doesn't do anything because `map` does not iterate by itself.